crates/ark/src/lsp/completions/function_context.rs

```diff
@@ -0,0 +1,228 @@
+//
+// function_context.rs
+//
+// Copyright (C) 2025 Posit Software, PBC. All rights reserved.
+//
+//
+
+use tower_lsp::lsp_types::Range;
+use tree_sitter::Node;
+
+use crate::lsp::document_context::DocumentContext;
+use crate::lsp::encoding::convert_tree_sitter_range_to_lsp_range;
+use crate::treesitter::node_find_parent_call;
+use crate::treesitter::node_text;
+use crate::treesitter::BinaryOperatorType;
+use crate::treesitter::NodeType;
+use crate::treesitter::NodeTypeExt;
+use crate::treesitter::UnaryOperatorType;
+
+/// Represents how a function is being used in an expression
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) enum FunctionUsage {
+    /// Function is being called, e.g., `foo()`
+    Call,
+    /// Function is being referenced without calling, e.g., `foo` in `debug(foo)`
+    Reference,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct FunctionContext {
+    /// The name of the function (could be, and often is, a fragment)
+    pub name: String,
+    /// The LSP range of the function identifier
+    pub range: Range,
+    /// How the function is being used (call vs reference)
+    pub usage: FunctionUsage,
+    /// The status of the function's arguments
+    pub arguments_status: ArgumentsStatus,
```

Comment:
No need to change now, but to me it feels like this should be part of a "call" struct.
Maybe that Call struct would be an argument to `FunctionRefUsage::Call()`.

This way we keep a clean distinction between identifiers and calls. This would avoid the somewhat awkward `Absent` argument status.